articles/mysql-query-performance-optimization-tips/index.md

```diff
@@ -0,0 +1,180 @@
+
+```sql
+SELECT * FROM city WHERE name LIKE '%Al%';
+```
+
+```sql
+SELECT * FROM city WHERE name LIKE 'Al%';
+```
+
+#### Specify Columns in SELECT Function
+SELECT * (select all) is used as a shortcut to query all columns available in a table.  This requires more resources than using a SELECT statement with only the columns you need for that specific query. For example, a customer table with 20 different columns and a hundred thousand entries. If you want to select a city with ID and Name only; use
+
+```sql
+SELECT ID, Name, District FROM city;
+```
+
+instead of
+
+```sql
+SELECT * FROM city;
+```
+
+The second example will take more time to run to completion.
+
+#### Avoid SELECT DISTINCT
+DISTINCT is used to remove duplicate rows with SELECT statements. The DISTINCT command requires more sorting and reading of the database, which requires more processing power. DISTINCT can be replaced with GROUP BY to get the same results. See the two queries below.
+
+```sql
+SELECT col1, col2 FROM table GROUP BY col1, col2;
+```
+
+```sql
+SELECT DISTINCT col1, col2 FROM table;
+```
+
+#### Use LIMIT
+Sometimes we need a specified number of rows from a result set. LIMIT clause is used in the query to specify the number of rows instead of fetching the whole result set. Fetching the entire result set requires more resources compared to fetching a specified number of rows. See the queries below, one without LIMIT, another one with the LIMIT clause.
+
+```sql
+SELECT ID, Name, District FROM city;
+```
+
+```sql
+SELECT ID, Name, District FROM city LIMIT 10;
+```
+
+#### MySQL Query Caching
+MySQL Query Caching provides database caching functionality. The SELECT statement text and the retrieved result are stored in the cache. When you make a similar query to the one already in the cache, MySQL will respond and give a query already in the cache. In this way, fewer resources are used, and your query runs faster.
+
+This works best with a database the more select queries are made. Once the table is updated, the cached query and result become invalid. Thus, caching may not work with an application that updates the table frequently.
+
+The command below is used to check if query cache is enabled in MySQL.
+
+```sql
+SHOW VARIABLES LIKE 'have_query_cache';
+```
+
+If the query cache is not set, set the query cache by following guidelines on [MySQL Documentation](https://dev.mysql.com/doc/refman/5.6/en/query-cache-configuration.html#:~:text=To%20set%20the%20size%20of,default%20for%20query_cache_type%20of%200.).
+
+#### Converting OUTER JOINs to INNER JOINs
+An INNER JOIN returns rows that contain columns from both tables. Unlike INNER JOIN, OUTER JOIN returns rows where no matches have been found on both tables. Therefore, OUTER JOIN does more work than inner join, increasing total execution time.
```

Comment:
inner join should be in all caps in order to conform to your established convention.